xdsl/dialects/builtin.py

```diff
@@ -460,7 +456,7 @@ def from_params(
 
 AnyVectorType: TypeAlias = VectorType[Attribute]
 
-_TensorTypeElems = TypeVar("_TensorTypeElems", bound=Attribute, covariant=True)
```

Comment:
I don't think this is the solution we want.
Since our attributes are immutable, this should be covariant.
I guess you changed this because the constructors need a different type variable, which is contravariant right?

Reply:
I decided to revert this change for the time being, s.t. we can get this PR merged asap. Let's look at this today!